utils/proxy/proxyconfig.go

```diff
@@ -163,6 +163,9 @@ var portMap = map[string]string{
 func canonicalAddr(url *url.URL) string {
 	addr := url.Host
 	if !hasPort(addr) {
+		if strings.HasPrefix(addr, "[") && strings.HasSuffix(addr, "]") {
```

Comment:
this feels really bad.
Why wouldn't JoinHostPort be aware of IPv6 addresses and do the right thing?